test/basic_any_test_rv.cpp

```diff
@@ -0,0 +1,331 @@
+//  Unit test for boost::basic_any<>.
+//
+//  See http://www.boost.org for most recent version, including documentation.
+//
+//  Copyright Antony Polukhin, 2013-2019.
+//
+//  Distributed under the Boost
+//  Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt).
+
+#include <cstdlib>
+#include <string>
+#include <utility>
+
+#include <boost/basic_any.hpp>
+#include "test.hpp"
+#include <boost/move/move.hpp>
+
+#ifdef BOOST_NO_CXX11_RVALUE_REFERENCES
+
+int main() 
+{
+    return EXIT_SUCCESS;
+}
+
+#else 
+
+namespace any_tests
+{
+    typedef test<const char *, void (*)()> test_case;
+    typedef const test_case * test_case_iterator;
+
+    extern const test_case_iterator begin, end;
+}
+
+int main()
+{
+    using namespace any_tests;
+    tester<test_case_iterator> test_suite(begin, end);
+    return test_suite() ? EXIT_SUCCESS : EXIT_FAILURE;
+}
+
+namespace any_tests // test suite
```

Comment:
Hmmm. Too much copy-pasting.

Change this file to a header, convert the any_tests to a templated structure with template parameter `class Any`. Use that header for testing the old `boost::any` and the new one.

Reply:
I completely agree that we need to avoid code duplication in this place. I made the common header just for move (rv) tests for both basic_any and any to verify the approach. The only thing I don't like there and the ANY_CAST_OP define. I faced with problem to pass "basic_any_cast" or "any_cast" without higher code modifications when templatized the common class. If we are ok with the approach I may also consider to do the common header for another big test. Otherwise let's think about something better.